File: Scripts/laptop/factorial/go/factorial.go
package main


func factorial(x uint) uint {
	if x == 0 {
		return 1
	}
	
	return x * factorial(x-1)
}

func executeTask(i int) int {
	a := uint(0)
	x := uint(10)
        a = factorial(x)
	_=a
	return i + 1;
}

func main() {
	r := int(1);

 	for i := 0; i < 2000000000; i++ {
		r = executeTask(i + r)
	}
}

File: Scripts/laptop/factorial/perl/factorial.pl

my $r = 1;

sub executeTask {
	my ($i) = @_;
	factorial( 10 );
	return ($i + 1);
}

foreach ( $a = 0; $a < 2000000000; $a + $a + 1 ) {
	$r = executeTask($a + $r);
}

sub factorial
{
   my $number = shift;   # get the argument

   if ( $number <= 1 ) { # base case
      return 1;
   }
   else {                # recursive step
      return $number * factorial( $number - 1 );
   }
}
